### ISSUE TITLE
Be able to become a libretro core

### ISSUE THREAD

**Issue by rhaleblian** [linappleii/linapple#41]
it's always the new guy who comes into the room with the crazy idea.

1. i wanted to remap joystick controls.
2. i wanted to remap keyboard controls. devs in the 80's didn't have any conventions to guide what keys they chose. and i think they didn't have up arrow and down arrow? !!!
3. i wanted to run under macOS.
4. i wanted to be compatible with applewin to exploit their non-Windows-specific features. i saw that applewin code is very Windows-y.

a core would give you 1-3 and change the playing field re 4.

**Comment by rhaleblian** [comment]
i have never written a core, so looked at the Vectrex source and quickly made my own core that just draws random stuff on screen. i'd say that porting would be doable but not trivial. you'd want to refactor out all the event, input and draw code so that you can ask for an execution step and a framebuffer for it.

**Comment by ghedger** [comment]
It's a worthy concept.  It may be viable to use a pared-down derivative forked from the linapple code base.

**Comment by rhaleblian** [comment]
Exactly. Looking from the other direction, I was looking at AppleWin and there is a lot of Windows code now, and imperfect separation.

**Comment by audetto** [comment]
This might not be an answer to the original issue, but I made a libretro core based on AppleWin here https://github.com/audetto/AppleWin/blob/master/linux.md#ra2

**Comment by maxolasersquad** [comment]
I'm doing a lot of work in the [sdl3](https://github.com/linappleii/linapple/tree/sdl3) branch that should put us in very good shape for Libretro.

**Comment by maxolasersquad** [comment]
I _think_ it is good now, but I've still got a lot more work to do. When done I'll see about reaching out to that team to make sure the code works for LibRetro.
